sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -32,6 +100,412 @@ export class StreamingReceiver extends MessageReceiver {
+          error.code !== ConditionErrorNameMapper["com.microsoft:message-lock-lost"] &&
+          this.receiveMode === ReceiveMode.peekLock &&
+          this.isOpen() // only try to abandon the messages if the connection is still open
+        ) {
+          try {
+            log.error(
+              "[%s] Abandoning the message with id '%s' on the receiver '%s' since " +
+                "an error occured: %O.",
+              connectionId,
+              bMessage.messageId,
+              this.name,
+              error
+            );
+            await bMessage.abandon();
+          } catch (abandonError) {
+            const translatedError = translate(abandonError);
+            log.error(
+              "[%s] An error occurred while abandoning the message with id '%s' on the " +
+                "receiver '%s': %O.",
+              connectionId,
+              bMessage.messageId,
+              this.name,
+              translatedError
+            );
+            this._onError!(translatedError);
+          }
+        }
+        return;
+      } finally {
+        this._receiverHelper.addCredit(1);
+      }
+
+      // If we've made it this far, then user's message handler completed fine. Let us try
+      // completing the message.
+      if (
+        this.autoComplete &&
+        this.receiveMode === ReceiveMode.peekLock &&
+        !bMessage.delivery.remote_settled
+      ) {
+        try {
+          log[this.receiverType](
+            "[%s] Auto completing the message with id '%s' on " + "the receiver '%s'.",
+            connectionId,
+            bMessage.messageId,
+            this.name
+          );
+          await bMessage.complete();
+        } catch (completeError) {
+          const translatedError = translate(completeError);
+          log.error(
+            "[%s] An error occurred while completing the message with id '%s' on the " +
+              "receiver '%s': %O.",
+            connectionId,
+            bMessage.messageId,
+            this.name,
+            translatedError
+          );
+          this._onError!(translatedError);
+        }
+      }
+    };
+  }
+
+  private _getHandlers(): ReceiverHandlers {
+    return {
+      onMessage: (context: EventContext) =>
+        this._onAmqpMessage(context).catch(() => {
+          /* */
+        }),
+      onClose: (context: EventContext) =>
+        this._onAmqpClose(context).catch(() => {
+          /* */
+        }),
+      onSessionClose: (context: EventContext) =>
+        this._onSessionClose(context).catch(() => {
+          /* */
+        }),
+      onError: this._onAmqpError,
+      onSessionError: this._onSessionError
+    };
+  }
+
+  stopReceivingMessages(): Promise<void> {
+    return this._receiverHelper.stopReceivingMessages();
+  }
+
+  init(useNewName: boolean, abortSignal?: AbortSignalLike): Promise<void> {
+    const options: ReceiverOptions = this._createReceiverOptions(useNewName, this._getHandlers());
```

Comment:
Nit: We shouldn't need the type declaration here. Type can be inferred by the output of the _create... method

```suggestion
    const options = this._createReceiverOptions(useNewName, this._getHandlers());
```

Reply:
Yes, removed.